include/openthread/nat64.h

```diff
@@ -88,6 +88,127 @@ typedef struct otIp4Cidr
     uint8_t      mLength;
 } otIp4Cidr;
 
+/**
+ * @struct otNat64Counters
+ *
+ * This structure represents the counters for NAT64.
+ *
+ */
+typedef struct otNat64Counters
+{
+    uint64_t m4To6SumPackets;
+    uint64_t m4To6SumBytes;
+    uint64_t m6To4SumPackets;
+    uint64_t m6To4SumBytes;
+} otNat64Counters;
+
+/**
+ * @struct otNat64ProtocolCounters
+ *
+ * This structure represents the counters for the protocols supported by NAT64.
+ *
+ */
+typedef struct otNat64ProtocolCounters
+{
+    otNat64Counters mTotal;
+    otNat64Counters mIcmp;
+    otNat64Counters mUdp;
+    otNat64Counters mTcp;
+} otNat64ProtocolCounters;
+
+/**
+ * Packet drop reasons.
+ *
+ */
+typedef enum otNat64DropReason
+{
+    OT_NAT64_DROP_REASON_UNKNOWN = 0,       ///< Packet drop for unknown reasons.
+    OT_NAT64_DROP_REASON_ILLEGAL_PACKET,    ///< Packet drop due to failed to parse the datagram.
+    OT_NAT64_DROP_REASON_UNSUPPORTED_PROTO, ///< Packet drop due to unsupported IP protocol.
+    OT_NAT64_DROP_REASON_NO_MAPPING, ///< Packet drop due to no corresponding mappings found, or address mapping pool
+                                     ///< exhausted.
+    //---
+    OT_NAT64_DROP_REASON_COUNT,
+} otNat64DropReason;
+
+/**
+ * @struct otNat64ErrorCounters
+ *
+ * This structure represents the counters of dropped packets due to errors when handling NAT64 packets.
+ *
+ */
+typedef struct otNat64ErrorCounters
+{
+    uint64_t mCount4To6[OT_NAT64_DROP_REASON_COUNT];
+    uint64_t mCount6To4[OT_NAT64_DROP_REASON_COUNT];
+} otNat64ErrorCounters;
+
+/**
+ * Gets the counters of the NAT64 translator, the counter is counted since the instance initialized.
+ *
+ * This function is available only when `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` is enabled.
+ *
+ * @param[in]      aInstance      A pointer to an OpenThread instance.
+ * @param[out]     aCounters      A pointer to an `otNat64Counters` where the counters of NAT64 translator will be
+ * placed.
+ *
+ */
+void otNat64GetCounters(otInstance *aInstance, otNat64ProtocolCounters *aCounters);
+
+/**
+ * Gets the counters of errors during processing packets by the NAT64 translator, the counter is counted
+ * since the instance initialized.
+ *
+ * This function is available only when `OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE` is enabled.
+ *
+ * @param[in]      aInstance      A pointer to an OpenThread instance.
+ * @param[out]     aCounters      A pointer to an `otNat64Counters` where the counters of NAT64 translator will be
+ * placed.
+ *
+ */
+void otNat64GetErrorCounters(otInstance *aInstance, otNat64ErrorCounters *aCounters);
+
+/**
+ * @struct otNat64AddressMapping
+ *
+ * This structure represents an address mapping record for NAT64.
+ *
+ */
+typedef struct otNat64AddressMapping
+{
+    uint64_t mId; // The unique id for a mapping session.
+
+    otIp4Address mIp4;
+    otIp6Address mIp6;
+    uint32_t     mExpiry;
```

Comment:
Change to return the remaining time (in msec) instead of absolute time (relative to `AlarmMilliGetNow()`)